api/src/main/java/org/apache/iceberg/SplittableScanTask.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+public interface SplittableScanTask<ThisT> extends ScanTask {
```

Comment:
Do we need this to be SplittableScanTask since I think our only usage is in combining? I do like the idea here though I just wonder if we will actually need to re-split or if we should always be producing minimally sized splits (one row group) and then combining them?

Avro is still an issue I guess ...

Reply:
We actually leverage `split` before combining in `TableScanUtil`.

Avro is still splittable, right? I am not sure whether we persist split offsets in the metadata so planning may not be as optimal as for Parquet.